docs/storage-config.md

```diff
@@ -0,0 +1,267 @@
+against the master storage.
+
+It is, however, also valid to want local file / block device based storage for
+the virtual machine, whose content is initialized by the master referenced
+storage. This can be achieved by introducing the notion of a "copy mode" in the
+disk backend configuration. This would accept the following options
+
+* None: the storage is directly accessed by QEMU. This is the default.
+* Copy-on-write: an overlay is provided with copy-on-write semantics against the
+  master storage. The local overlay will only contain changes made by the guest.
+* Copy-on-read: an overlay is provided with copy-on-read semantics against the
+  master storage. The local overlay will contain both changes made by the guest,
+  and copies of any blocks read.
+* Clone: A full copy of the master storage is made to a local file. Once this is
+  created, no further use is made of the master storage.
+
+The first copy mode provides long term persistent storage, while the other modes
+provide ephemeral storage that is discarded when the virtual machine is deleted
+from the node.
+
+The host needs to have somewhere to store the local overlays / files. This is
+achieved by configuring one or more volumes for the libvirtd POD when it is
+created, and mounting them into a defined location. Any valid PVC is usable for
+this purpose. If the PVC backing store is only accessible for a single host,
+then when performing migration, QEMU block storage migration feature will need
+to be enabled. If the PVC references a networked filesystem that can be mounted
+on multiple hosts, then VMs can migrate with no need to copy emphemeral disks.
+If multiple volume mounts are made available to the libvirtd POD, annotations
+would need to be used to tag them for different usage types. This might be useful
+if a cluster needs to provide ephemeral storage with differing performance grades
+or costs (eg one volume provides cheap & slow rotating rust, while another volume
+provides expensive & fast solid state). The VM spec disk configuration would be
+able to optionally include a reference to the usage type it desires, to allow
+virt-handler to map the disk to the particular volume mount.
+
+Thus the libvirtd container may contain
+
+```yaml
+    annotations:
+      kubevirt.io/storage.usage-type.fast: /storage/ssd
+      kubevirt.io/storage.usage-type.slow: /storage/rust
+    volumeMounts:
+      - mountPath: /storage/ssd
+        name: storage-ssd
+      - mountPath: /storage/rust
+        name: storage-rust
+```
+
+NB The DaemonSet resource does not provide a way to dynamically assign
+different PVCs to each libvirtd POD it creates. Thus in the short term, to make
+use of this feature would require the cloud admin to manually create libvirtd
+POD specs for each compute node, instead of using a DeamonSet resource. It is
+anticipated that either DaemonSet would be enhanced eventually, or replaced by
+a more advanced resource that can dynamically assigns PVCs to each POD.
```

Comment:
There is no special binding between a pod and a PVC. A user just creates and deletes PVCs when needed. virt-handler has to take the access modes into account, and once the PVC is not needed anymore, one deletes it and the dynamic provisioner will recycle the associated PV. There is not much more involved.